src/main/java/roomescape/member/infra/MemberHandlerInterceptor.java

```diff
@@ -0,0 +1,41 @@
+package roomescape.member.infra;
+
+import jakarta.servlet.http.Cookie;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+import roomescape.member.application.AuthService;
+import roomescape.member.dto.MemberResponseDto;
+
+import java.io.IOException;
+
+import static roomescape.member.infra.TokenUtil.extractTokenFromCookie;
+
+@Component
+public class MemberHandlerInterceptor implements HandlerInterceptor {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MemberHandlerInterceptor.class);
+    private final AuthService authService;
+
+    public MemberHandlerInterceptor(AuthService authService) {
+        this.authService = authService;
+    }
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws IOException {
+        final Cookie[] cookies = request.getCookies();
+        final String token = extractTokenFromCookie(cookies);
+        final MemberResponseDto member = authService.findMember(token);
+
+        LOGGER.info("check role of member : {}", member);
+
+        if (member == null || !member.getRole().equals("ADMIN")) {
```

Comment:
```suggestion
        if (member == null || !"ADMIN".equals(member.getRole())) {
```

소소한 디테일인데, 이 로직을 위와 같이 작성하면 어떤 장점이 있을까요? :)

Reply:
member.getRole()은 해당 값이 null 일경우 NPE 가 발생할 수 있는데.
후자의 경우 member.getRole()이 null 이면, false를 반환해 NPE가 발생하지 않네요 :)

이렇게 또 하나 배워가네요 ㅎㅎ

```java
public class test {

    public static void main(String[] args) {
        String answer = "hi";
        System.out.println("hi".equals(answer));

        String answer2 = null;
        System.out.println(answer2.equals("hi"));
    }
}
```

![image](https://github.com/next-step/spring-roomescape-auth/assets/105576721/1a4dcfe1-8b6f-4fb4-b407-fe7bb15f0971)